grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -522,10 +522,12 @@ class StockMovementService {
                     editPageItem << [quantityOnHandRequesting: quantityOnHandRequesting]
                     if (requisition.requisitionTemplate) {
                         def stocklist = Requisition.get(requisition.requisitionTemplate.id)
-                        def quantityOnStocklist = stocklist.requisitionItems?.find { it.product == editPageItem.product && it.orderIndex * 100 == editPageItem.sortOrder }?.quantity?:0
+                        def quantityOnStocklist = stocklist.requisitionItems?.find {
+                            it.product == editPageItem.product && (it.orderIndex * 100 == editPageItem.sortOrder || it.orderIndex == editPageItem.sortOrder)
```

Comment:
@DominikaHatala I am not happy with this way of finding it, but for now, it should work. It could be refactored in the next sprint.

Reply:
@awalkowiak That is the only thing that doesn't look nice for me in this PR but as you said it can be refactored later so I'm approving.